File: docs/guides/dart/lib/setUpTransformationOptionsWithOverrides.dart
import 'package:algolia_client_search/algolia_client_search.dart';

void main() async {
  // Override the Ingestion API defaults. Any option you don't set keeps its default.
  // Replace 'us' with 'eu' if your Algolia application uses the Europe analytics region.
  final client = SearchClient(
    appId: 'ALGOLIA_APPLICATION_ID',
    apiKey: 'ALGOLIA_API_KEY',
    transformationOptions: TransformationOptions(
      region: 'us',
      ingestionClientOptions: ClientOptions(
        connectTimeout: Duration(seconds: 5),
        readTimeout: Duration(seconds: 30),
      ),
    ),
  );

  try {
    // Save records, transforming them through the Push connector
    await client.saveObjectsWithTransformation(
      indexName: "<YOUR_INDEX_NAME>",
      objects: [
        {
          'objectID': "1",
          'name': "Adam",
        },
        {
          'objectID': "2",
          'name': "Benoit",
        },
      ],
      waitForTasks: true,
    );
    print('Done!');
  } catch (e) {
    print('Error: ${e.toString()}');
  }
}

File: docs/guides/ruby/setUpTransformationOptionsWithOverrides.rb
require "algolia"

# Override the Ingestion API defaults. Any option you don't set keeps its default.
# Replace 'us' with 'eu' if your Algolia application uses the Europe analytics region.
client = Algolia::SearchClient.with_transformation(
  "ALGOLIA_APPLICATION_ID",
  "ALGOLIA_API_KEY",
  Algolia::TransformationOptions.new("us", connect_timeout: 5000, read_timeout: 30000)
)

# Save records, transforming them through the Push connector
client.save_objects_with_transformation(
  "<YOUR_INDEX_NAME>",
  [{objectID: "1", name: "Adam"}, {objectID: "2", name: "Benoit"}],
  true
)

puts("Done!")
